include/sacn/cpp/source.h

```diff
@@ -155,6 +155,8 @@ class Source
     /** Create an empty, invalid data structure by default. */
     UniverseNetintList() = default;
     UniverseNetintList(sacn_source_t source_handle, uint16_t universe_id);
+    UniverseNetintList(sacn_source_t source_handle, uint16_t universe_id,
+                       const std::vector<SacnMcastInterface>& network_interfaces);
```

Comment:
The thought behind this was to enable the application to initialize the whole vector using list initialization. I end up doing this in the unit tests:
`static std::vector<sacn::Source::UniverseNetintList> kTestNetintLists = {{kTestHandle, kTestUniverse, kTestNetints},
                                                                         {kTestHandle, kTestUniverse2, kTestNetints},
                                                                         {kTestHandle2, kTestUniverse, kTestNetints},
                                                                         {kTestHandle2, kTestUniverse2, kTestNetints}};`

If I wanted to I could even define the netints individually as well:
`static std::vector<sacn::Source::UniverseNetintList> kTestNetintLists = {
    {kTestHandle, kTestUniverse, {{{kEtcPalIpTypeV4, 1u}, kEtcPalErrOk}, {{kEtcPalIpTypeV4, 2u}, kEtcPalErrOk}}},
    {kTestHandle, kTestUniverse2, {{{kEtcPalIpTypeV4, 3u}, kEtcPalErrOk}}},
    {kTestHandle2, kTestUniverse, {{{kEtcPalIpTypeV4, 5u}, kEtcPalErrOk}, {{kEtcPalIpTypeV4, 6u}, kEtcPalErrOk}}},
    {kTestHandle2, kTestUniverse2, {{{kEtcPalIpTypeV4, 7u}, kEtcPalErrOk}}}};`

If the UniverseNetintList constructor is removed, then the list initialization no longer builds, and instead results in `error C2440: 'initializing': cannot convert from 'initializer list' to 'std::vector<sacn::Source::UniverseNetintList,std::allocator<_Ty>>'`.

So I kept the constructor in because I could see real use cases that use list initialization as a way to initialize this vector, without needing to initialize each element individually and calling push_back for each. It could be that this approach doesn't end up getting used much though.

Reply:
I added some documentation to clarify this: https://github.com/ETCLabs/sACN/pull/48/commits/7a6c9cc7ff09308ccd821de3e4acf9e34468dce3